### PR TITLE
Add color picker to select tree color

### LINKED ISSUE #4794: User-definable colors for trees
Double-click on a tree in the tree list to show a color palette.

### PR DESCRIPTION
**Pull request by MichaelBuessemeyer**
### URL of deployed dev instance (used for testing):
- https://colorselectionfortrees.webknossos.xyz

### Steps to test:
- Open a skeleton tracing and create a tree with some nodes.
- Then use the color selection by clicking the [eye-dropper icon](https://fontawesome.com/icons/eye-dropper?style=solid).
The color of the tree should now be the newly selected color.

### Issues:
- fixes #4794

------
- [x] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- ~~[ ] Updated [(unreleased) migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable~~
- ~~[ ] Updated [documentation](../blob/master/docs) if applicable~~
- ~~[ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes~~
- ~~[ ] Needs datastore update after deployment~~
- [x] Ready for review

### CODE REVIEW

**Review by daniel-wer — COMMENTED** [review]
Nice! I left only a few suggestions.

I think we can improve the UI for this a bit, because the functionality is kind of scattered right now and I feel like the color picking is too prominent. Some options, like "Measure skeleton length" and "Shuffle tree color" are in the dropdown menu up top and this color picker is next to the tree name and always visible. We could put all of these three things into a settings dropdown next to the tree name in the same way it's done for the groups. These setting would also only appear when hovering to keep the UI more clear. Maybe @philippotto has some opinion or suggestions for this as well? :)

**Comment by philippotto** [comment]
> I think we can improve the UI for this a bit, because the functionality is kind of scattered right now and I feel like the color picking is too prominent. Some options, like "Measure skeleton length" and "Shuffle tree color" are in the dropdown menu up top and this color picker is next to the tree name and always visible. We could put all of these three things into a settings dropdown next to the tree name in the same way it's done for the groups. These setting would also only appear when hovering to keep the UI more clear. Maybe @philippotto has some opinion or suggestions for this as well? :)

Yes, I completely agree with you :+1: It would also be cool if one could right click on a tree and then the settings submenu would open, too (same as clicking on the cog icon). I think antd supports `contextmenu` as a trigger for opening the menu.

**Comment by MichaelBuessemeyer** [comment]
> I think we can improve the UI for this a bit, because the functionality is kind of scattered right now and I feel like the color picking is too prominent. Some options, like "Measure skeleton length" and "Shuffle tree color" are in the dropdown menu up top and this color picker is next to the tree name and always visible. We could put all of these three things into a settings dropdown next to the tree name in the same way it's done for the groups. These setting would also only appear when hovering to keep the UI more clear. 

I also think this a very good idea :+1:

File: frontend/javascripts/oxalis/view/right-menu/tree_hierarchy_view.js [hunk 1]
```diff
@@ -362,6 +366,34 @@ class TreeHierarchyView extends React.PureComponent<Props, State> {
...
+              style={{
+                position: "absolute",
+                top: "50%",
+                left: "50%",
+                transform: "translate(-50%, -50%)",
+                cursor: "pointer",
+              }}
+            />
+            <input
+              type="color"
```
[the remark this answers is not in the document]
Comment (MichaelBuessemeyer) [reply frontend/javascripts/oxalis/view/right-menu/tree_hierarchy_view.js]: This sounds like a good idea but when I implemented this, having a separate color for a single icon does look good in my opinion:
Screenshot_20201106_144220
Screenshot_20201106_144630

What do you think about this?

[the remark this answers is not in the document]
Comment (daniel-wer) [reply frontend/javascripts/oxalis/view/right-menu/tree_hierarchy_view.js]: @MichaelBuessemeyer Sorry, that's not what I meant. I meant the color that is selected by default when the color picker is opened:
color_picker
which is currently always black. By setting the value of the color input (see the link from my earlier comment) it should be possible to initialize the color correctly.

File: frontend/javascripts/oxalis/view/right-menu/tree_hierarchy_view.js [hunk 2]
```diff
@@ -353,6 +365,58 @@ class TreeHierarchyView extends React.PureComponent<Props, State> {
...
+          <Menu.Item key="changeTreeColor">
+            <div style={{ position: "relative", display: "inline-block", width: "100%" }}>
+              <i
+                className="fas fa-eye-dropper fa-sm"
+                style={{
+                  cursor: "pointer",
+                  color: Utils.rgbToHex(Utils.map3(value => value * 255, tree.color)),
+                }}
+              />{" "}
+              Change the color
```
Comment (daniel-wer) [inline frontend/javascripts/oxalis/view/right-menu/tree_hierarchy_view.js:379]: In accordance with the next menu entry "Shuffle Tree Color", I would propose to name this
```suggestion
              Select Tree Color
```

**Review by daniel-wer — COMMENTED** [review]
Thanks for improving the UX by adding the tree setting dropdowns :+1: 

Please see my comment regarding the default color that is selected when the color pickers are opened (that I added before this review).
Also, the color picker no longer works for me as it instantly closes after clicking the menu entry (is only a problem in Chrome - Firefox works).

File: frontend/javascripts/oxalis/view/right-menu/tree_hierarchy_view.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (MichaelBuessemeyer) [reply frontend/javascripts/oxalis/view/right-menu/tree_hierarchy_view.js]: Got it :) This makes way more sense :+1:

**Comment by MichaelBuessemeyer** [comment]
>Also, the color picker no longer works for me as it instantly closes after clicking the menu entry (is only a problem in Chrome - Firefox works). 

This took me quite some time to figure it out.
I tried using a library for this but the library made it only worse.


The solution: Somehow controlling the dropdown menu's visibility made it work. 

I think the actual problem was that the color selection is a child of the dropdown menu and as the color picker opens, the dropdown menu loses focus and collapses which also closes the color picker.

**Comment by MichaelBuessemeyer** [comment]
@daniel-wer I think I fixed all your mentions and the code looks fine. Could you please check the newest changes again?

**Review by daniel-wer — APPROVED** [review]
Very nice, works like a charm for me! :+1:

**Pull request merged**